staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -34,6 +35,9 @@ type logger interface {
 	Addf(format string, data ...interface{})
 }
 
+// respLoggerContextKey is used to store the respLogger pointer in the request context.
+type respLoggerContextKey struct{}
```

Comment:
The use of `struct{}` here makes me a bit nervous.  IIRC two `struct{}`s can have the same address which could maybe compare as equal?  The pattern we use throughout the codebase is:

```go
type requestInfoKeyType int

const requestInfoKey requestInfoKeyType = iota
```

Reply:
How is this different to my struct? iota gives an `0:int`.

Reply:
https://goplay.space/#Xe_YV8Iq7CT

Reply:
@enj switched to `iota`.